allennlp/run.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any, Dict, Union
```

Comment:
We need a place to put code that puts everything together and actually runs something.  The `training` module is _given_ a dataset, model, and so on, do the training loop.  This piece of code says, build all of the prerequisites, load data, do whatever, _then_ call stuff in `training`.  Other, similar pieces of code will build prerequisites for some other task, like loading a saved model and a test dataset, then evaluating the model.

My big question is how should we structure this code?  There are a few options:

1. We could have a bunch of scripts like this in some directory, not sure where (`allennlp/experiments/`?  `allennlp/drivers/`?).  These have code to actually run the operations you want, and get called by some CLI somewhere.

2. We could use a `Driver` abstraction, like I originally had this PR, and have each of these be a class instead of a script, with a `run()` method that does it, and a unified CLI into it.

3. DeepQA had a python file called `deep_qa/run.py` that had the top-level training and evaluation entry points, and that's where this particular file pattern came from.  We could keep this general idea, but change the name of the file...

4. Some other option?

I'm really not sure what the best way to go is.  Any thoughts?